File: polimorfismo.py
class Animal:
    def sonido(self):
        print("El animal hace un sonido")

class Perro(Animal):
    def sonido(self):
        print("El perro ladra")

class Gato(Animal):
    def sonido(self):
        print("El gato maulla")

a1 = Perro()
a2 = Gato()

a1.sonido()
a2.sonido()

File: polimorfismo.java
class Animal{
    public void sonido(){
        System.out.println("El animal hace un sonido");
    }
}

class Perro extends Animal{
    public void sonido(){
        System.out.println("El perro ladra");
    }
}

class Gato extends Animal{
    public void sonido(){
        System.out.println("El gato maulla");
    }
}

public class Main{
    public static void main(String[] args){
        Animal a1 = new Perro();
        Animal a2 = new Gato();

        a1.sonido();
        a2.sonido();
    }
}
